internal/util/ssa/metrics.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssa
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+func init() {
+	// Register the metrics at the controller-runtime metrics registry.
+	ctrlmetrics.Registry.MustRegister(requestTotal)
+}
+
+var (
+	// requestTotal reports request results.
+	requestTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "capi_ssa_cache_request_total",
+		Help: "Total number of ssa cache hit and miss requests.",
+	}, []string{
+		"status", "kind", "cache_owner",
```

Comment:
Let's use "controller" instead of "cache_owner". This aligns nicely with a bunch of controller-runtime metrics.

Reply:
Hm maybe we should do the same as CoreDNS. They simply have:
* coredns_cache_hits_total
* coredns_cache_misses_total

WDYT?

Reply:
sure whichever you think is best, i was using an example convention from this doc: https://kubernetes.io/docs/reference/instrumentation/metrics/

> `authentication_token_cache_request_total`
> - Stability Level:ALPHA
> - Type: Counter
> - Labels:`status`

Reply:
I personally think the CoreDNS style is more intuitive

Reply:
But I'm not an expert, if one of them clearly fits best practices and the other doesn't I'm fine with picking that one :)

Reply:
sounds good to keep in line w/controller-runtime 👍